### PR TITLE
Major refactor and reorganization of types

### PR DESCRIPTION
**Pull request by Slendy**
This PR reorganizes most of the type classes and enums within Lighthouse. It organizes all of the DbSet entity types into a single folder and other types are organized into ProjectLighthouse/Types. Some types are only used by a single server and as such, they exist in the corresponding server's Types folder.

### CODE REVIEW

File: ProjectLighthouse/Database.cs [hunk 1]
```diff
@@ -3,56 +3,56 @@
...
+using LBPUnion.ProjectLighthouse.Types.Misc;
 using Microsoft.AspNetCore.Http;
 using Microsoft.EntityFrameworkCore;
 
 namespace LBPUnion.ProjectLighthouse;
 
 public class Database : DbContext
 {
+    // ReSharper disable once InconsistentNaming
+    public DbSet<ApiKey> APIKeys { get; set; }
```
Comment (Zaprit) [inline ProjectLighthouse/Database.cs:25]: the variable should also be called ApiKeys, no?

Comment (Slendy) [reply ProjectLighthouse/Database.cs:25]: I tried to avoid doing a migration in this PR so it has to stay capitalized to match the actual table name

Comment (Zaprit) [reply ProjectLighthouse/Database.cs:25]: fair enough

**Review by Zaprit — COMMENTED** [review]
Pretty cool, that was a lot of files

**Comment by Slendy** [comment]
LGTM
image

**Pull request merged**